src/app/services/course.service.ts

```diff
@@ -1,9 +1,37 @@
 import { Injectable } from '@angular/core';
+import * as courseList from '../mocks/courses.json';
+import { ICourse } from '../modules/dashboard/interfaces/course.interface';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CourseService {
 
-  constructor() { }
+  public courses: ICourse[];
+
+  constructor() {
+    this.courses = (courseList as any).default;
+  }
+
+  public getList(): ICourse[] {
+    return this.courses;
+  }
+
+  public createCourse(course: ICourse): void {
+    this.courses.concat({...course});
```

Comment:
or `this.courses.push(course)`?

Reply:
this will mutate the data, I am trying not to. Why is it better to use push()?